presto-jdbc/src/main/java/io/prestosql/jdbc/ConnectionProperties.java

```diff
@@ -378,4 +380,42 @@ public ExtraCredentials()
             return ImmutableList.of(name, value);
         }
     }
+
+    private static class SessionProperties
+            extends AbstractConnectionProperty<Map<String, String>>
+    {
+        private static final Splitter NAME_PARTS_SPLITTER = Splitter.on('.');
+
+        public SessionProperties()
+        {
+            super("sessionProperties", NOT_REQUIRED, ALLOWED, SessionProperties::parseSessionProperties);
+        }
+
+        // Session properties consists of a list of session property name value pairs.
+        // E.g., `jdbc:presto://example.net:8080/?sessionProperties=abc:xyz;catalog.foo:bar` will create session properties `abc=xyz` and `catalog.foo=bar`
+        public static Map<String, String> parseSessionProperties(String sessionProperties)
+        {
+            return MAP_ENTRIES_SPLITTER.splitToList(sessionProperties).stream()
+                    .map(SessionProperties::parseSessionProperty)
+                    .collect(toImmutableMap(entry -> entry.get(0), entry -> entry.get(1)));
+        }
+
+        public static List<String> parseSessionProperty(String sessionProperty)
+        {
+            List<String> nameValue = MAP_ENTRY_SPLITTER.splitToList(sessionProperty);
+            checkArgument(nameValue.size() == 2, "Malformed session property: %s", sessionProperty);
```

Comment:
Can session property contain a colon?

(Look, since `MAP_ENTRY_SPLITTER` is opaque constant, i need to _remember_ it splits on colon, to ask myself this question. Extracting this as as a const makes the code less readable and doesn't remove any complexity.)